lib/modules/keyboardNav.js

```diff
@@ -1078,16 +1073,29 @@ module.toggleChildren = function() {
 	click(thisToggle);
 };
 
-module.toggleExpando = function() {
-	const selected = SelectedEntry.selectedThing();
-	if (!selected) return;
+module.toggleExpandos = function(scrollOnExpanding) {
+	const thing = SelectedEntry.selectedThing();
+	if (!thing) return;
+
+	const expandos = thing.getExpandos();
+	if (!expandos.length) return;
 
-	const thisExpando = selected.getExpandoButton();
-	if (thisExpando) {
-		const expanding = !thisExpando.classList.contains('expanded');
-		click(thisExpando);
-		if (module.options.scrollOnExpando.value && expanding) {
-			scrollToElement(selected.thing, { scrollStyle: 'top' });
+	// If any open expandos exists within thing, collapse all
+	// Else, expand all
+	const openExpandos = expandos.filter(v => v.open);
+
+	if (openExpandos.length) {
+		for (const expando of openExpandos) expando.collapse();
+	} else {
+		for (const expando of expandos) {
+			if (expando.constructor.name !== 'Expando' || expando::ShowImages.isExpandWanted(thing.thing)) {
```

Comment:
I'd prefer importing `Expando` and using `instanceof`, or checking for `expando.expand` (if that's equivalent)